keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -53,22 +66,22 @@ class ArtifactListener(
       artifactRepository.register(artifact)
     }
 
-    if (artifactRepository.versions(artifact).isEmpty()) {
-      storeLatestVersion(artifact, event.statuses)
+    if (artifact.type == DEB && artifactRepository.versions(artifact).isEmpty()) {
+      storeLatestDebVersion(artifact, event.statuses)
     }
   }
 
   /**
    * Grab the latest version which matches the statuses we care about, so the artifact is relevant.
    */
-  protected fun storeLatestVersion(artifact: DeliveryArtifact, statuses: List<ArtifactStatus>) =
+  protected fun storeLatestDebVersion(artifact: DeliveryArtifact, statuses: List<ArtifactStatus>) =
```

Comment:
Doesn't need to change in this PR, but instead of deb vs. docker, we should think about artifact formats in terms of those that need to be baked, vs. those that are directly deployable. We'll probably need to onboard RPM based use-cases in the near future, and possibly Chocolatey too. It's also interesting to consider rosco's support for baking various types of k8s manifests. In the near term, this may just mean thinking about things in terms of `Artifact.needsBake()` instead of `Artifact.isDeb()`.

Reply:
@asher Side-question: for the K8S manifests support in rosco, what's the output of "baking" them? Just curious.

Reply:
Oh that's a great point. I'll think about that going forward.